dbms/src/Storages/KVStore/MultiRaft/PrehandleSnapshot.cpp

```diff
@@ -163,6 +200,328 @@ PrehandleResult KVStore::preHandleSnapshotToFiles(
+        part_new_region,
+        index,
+        snaps,
+        proxy_helper,
+        tmt,
+        std::move(part_limit),
+        DM::SSTFilesToBlockInputStreamOpts(opt));
+    try
+    {
+        std::string limit_tag = part_limit.toDebugString();
+        auto [part_result, part_prehandle_result]
+            = executeTransform(part_new_region, prehandle_task, job_type, dm_storage, part_sst_stream, opt, tmt);
+        LOG_INFO(
+            log,
+            "Finished extra parallel prehandle task limit {} write cf {} lock cf {} default cf {} dmfiles {} error {}, "
+            "split_id={} region_id={}",
+            limit_tag,
+            part_prehandle_result.stats.write_cf_keys,
+            part_prehandle_result.stats.lock_cf_keys,
+            part_prehandle_result.stats.default_cf_keys,
+            part_prehandle_result.ingest_ids.size(),
+            magic_enum::enum_name(part_result.error),
+            extra_id,
+            part_new_region->id());
+        if (part_result.error == ReadFromStreamError::ErrUpdateSchema)
+        {
+            prehandle_task->store(true);
+        }
+        {
+            std::scoped_lock l(ctx->mut);
+            ctx->gather_res[extra_id] = std::move(part_result);
+            ctx->gather_prehandle_res[extra_id] = std::move(part_prehandle_result);
+        }
+    }
+    catch (Exception & e)
+    {
+        // The exception can be wrapped in the future, however, we abort here.
+        const auto & processed_keys = part_sst_stream->getProcessKeys();
+        LOG_INFO(
+            log,
+            "Parallel prehandling error {}"
+            " write_cf_off={}"
+            " split_id={} region_id={}",
+            e.message(),
+            processed_keys.write_cf,
+            extra_id,
+            part_new_region->id());
+        prehandle_task->store(true);
+        throw;
+    }
+}
+
+void executeParallelTransform(
+    LoggerPtr log,
+    ReadFromStreamResult & result,
+    PrehandleResult & prehandle_result,
+    const std::vector<std::string> & split_keys,
+    std::shared_ptr<DM::SSTFilesToBlockInputStream> sst_stream,
+    RegionPtr new_region,
+    DM::SSTFilesToBlockInputStreamOpts & opt,
+    const SSTViewVec & snaps,
+    const TiFlashRaftProxyHelper * proxy_helper,
+    TMTContext & tmt,
+    std::shared_ptr<std::atomic_bool> prehandle_task,
+    DM::FileConvertJobType job_type,
+    uint64_t index,
+    std::shared_ptr<StorageDeltaMerge> storage)
+{
+    using SingleSnapshotAsyncTasks = AsyncTasks<uint64_t, std::function<bool()>, bool>;
+    auto split_key_count = split_keys.size();
+    RUNTIME_CHECK_MSG(
+        split_key_count >= 1,
+        "split_key_count should be more or equal than 1, actual {}",
+        split_key_count);
+    LOG_INFO(
+        log,
+        "Parallel prehandling for single big region, range={}, split keys={}, region_id={}",
+        new_region->getRange()->toDebugString(),
+        split_key_count,
+        new_region->id());
```

Comment:
Move this logging to line 620 is better for reading

Reply:
Yes, but it will make `KVStore::preHandleSSTsToDTFiles` even longer, it takes already ~200 lines with deep indention